SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -817,3 +838,738 @@ void CMeshSolver::Restart_OldGeometry(CGeometry *geometry, CConfig *config) {
+          
+          cosTheta = cos(dtheta);  cosPhi = cos(dphi);  cosPsi = cos(dpsi);
+          sinTheta = sin(dtheta);  sinPhi = sin(dphi);  sinPsi = sin(dpsi);
+          
+          /*--- Compute the rotation matrix. Note that the implicit
+           ordering is rotation about the x-axis, y-axis, then z-axis. ---*/
+          
+          rotMatrix[0][0] = cosPhi*cosPsi;
+          rotMatrix[1][0] = cosPhi*sinPsi;
+          rotMatrix[2][0] = -sinPhi;
+          
+          rotMatrix[0][1] = sinTheta*sinPhi*cosPsi - cosTheta*sinPsi;
+          rotMatrix[1][1] = sinTheta*sinPhi*sinPsi + cosTheta*cosPsi;
+          rotMatrix[2][1] = sinTheta*cosPhi;
+          
+          rotMatrix[0][2] = cosTheta*sinPhi*cosPsi + sinTheta*sinPsi;
+          rotMatrix[1][2] = cosTheta*sinPhi*sinPsi - sinTheta*cosPsi;
+          rotMatrix[2][2] = cosTheta*cosPhi;
+          
+          for (iVertex = 0; iVertex < geometry->nVertex[iMarker]; iVertex++) {
+            
+            /*--- Index and coordinates of the current point ---*/
+            
+            iPoint = geometry->vertex[iMarker][iVertex]->GetNode();
+            Coord  = geometry->node[iPoint]->GetCoord();
+            
+            /*--- Calculate non-dim. position from rotation center ---*/
+            
+            for (iDim = 0; iDim < nDim; iDim++)
+              r[iDim] = (Coord[iDim]-Center[iDim])/Lref;
+            if (nDim == 2) r[nDim] = 0.0;
+            
+            /*--- Compute transformed point coordinates ---*/
+            
+            rotCoord[0] = rotMatrix[0][0]*r[0]
+                        + rotMatrix[0][1]*r[1]
+                        + rotMatrix[0][2]*r[2] + Center[0];
+            
+            rotCoord[1] = rotMatrix[1][0]*r[0]
+                        + rotMatrix[1][1]*r[1]
+                        + rotMatrix[1][2]*r[2] + Center[1];
+            
+            rotCoord[2] = rotMatrix[2][0]*r[0]
+                        + rotMatrix[2][1]*r[1]
+                        + rotMatrix[2][2]*r[2] + Center[2];
+            
+            /*--- Calculate delta change in the x, y, & z directions ---*/
+            for (iDim = 0; iDim < nDim; iDim++)
+              VarCoord[iDim] = (rotCoord[iDim]-Coord[iDim])/Lref;
+            if (nDim == 2) VarCoord[nDim] = 0.0;
+            
+            // /*--- Set node displacement for volume deformation ---*/
+            // geometry->vertex[iMarker][iVertex]->SetVarCoord(VarCoord);
+            
+            for (iDim = 0; iDim < 3; iDim++){
+              VarCoordAbs[iDim] = nodes->GetBound_Disp(iPoint, iDim) + VarCoord[iDim];
+            }
+            
+            nodes->SetBound_Disp(iPoint, VarCoordAbs);
+          }
+        }
+      }
+    }
+  }
+  /*--- For pitching we don't update the motion origin and moment reference origin. ---*/
+  
+}
+
+void CMeshSolver::Surface_Rotating(CGeometry *geometry, CConfig *config,
+                                        unsigned long iter, unsigned short iZone) {
```

Comment:
Any plans regarding reducing the duplication here? Maybe these functions will not be needed by the grid movement after this PR?

Reply:
The surface deformation calls aren't really part of this PR but have been merged in here from #833 to test with alternate problems. #833 has passed most checks but fails on CodeFactor precisely because of these routines which have been flagged as complex code. So I think it might be best to refactor there and remove unused code.